internal/gw/publish.go

```diff
@@ -144,3 +144,19 @@ func (p *publish) Commit(ctx context.Context) error {
 	err = task.Await(ctx)
 	return err
 }
+
+func (p *publish) Get(ctx context.Context) error {
```

Comment:
Ask for suggestion:

I am not sure if this `publish::Get` is a nice method name here, I think it should be `publish::GetPublish` so it can be consistent with `publish::NewPublish`, and update `gwClient::GetPublish` to another name, maybe `gwClient::MakePublish`.

Reply:
I think Publish.Get makes sense because it's returning itself. Client.GetPublish also makes sense because it's getting the publish that's associated with itself.
If anything, publish::NewPublish -> publish::New.

Reply:
I think publish::Get makes sense because it's returning the publish itself. gwClient::GetPublish also makes sense because it's returning the publish associated with the client. So, these seem okay to me.

If anything, publish::NewPublish should be publish::New.

Reply:
Looking at what it does, I don't think `Get` or `GetPublish` are good names.  It doesn't behave as one would normally expect a "getter" to behave:

- it doesn't return the object it "gets"
- you already need to have the publish object before you can call it. So why do you want to Get an object you already have anyway?

Judging from the implementation, the point of the method seems to be only to verify that the object exists and return an error otherwise.  Then, thinking from the point of view of someone reading code which calls the method, I think a better name would be something like EnsureExists.  Considering the following two examples:

```
                # example 1
		err = publish.Get(ctx)
		if err != nil { ... }
```


```
                # example 2
		err = publish.EnsureExists(ctx)
		if err != nil { ... }
```

It seems to me that the point of example 1 can be easily misunderstood as either "get a publish which is related to this one", or "refresh all fields on the publish to match the server's current state".  Both of which are wrong.

Example 2 on the other hand makes it clear why we are calling this method and what the behavior is expected to be.

It seems like a moot point though: I don't think the method should even exist, because now 100% of the time gwClient.GetPublish is called, you immediately call publish.Get afterward.  Isn't it clearer to just fold the functionality into gwClient.GetPublish?  i.e., change the behavior of that method so that it requires the publish to exist in order to succeed?